include/library/spdm_crypt_lib.h

```diff
@@ -971,6 +971,32 @@ bool libspdm_get_dmtf_subject_alt_name(const uint8_t *cert, const size_t cert_si
                                        size_t *name_buffer_size,
                                        uint8_t *oid, size_t *oid_size);
 
+/**
+ * This function verifies the integrity of certificate chain data without spdm_cert_chain_t header for a
+ * SET_CERTIFICATE
+ *
+ * This function also allows for not-validating the leaf certificate in the chain. This maybe useful for a
+ * requester to validate the immutable certificates ahead of a `SET_CERTIFICATE` request
+ * (i.e using the AliasCertificate model).
+ *
+ * @param  cert_chain_data          The certificate chain data without spdm_cert_chain_t header.
+ * @param  cert_chain_data_size      size in bytes of the certificate chain data.
+ * @param  base_asym_algo            SPDM base_asym_algo
+ * @param  base_hash_algo            SPDM base_hash_algo
+ * @param  is_requester_cert         Is the function verifying requester or responder cert.
+ * @param  is_device_cert_model      If true, the cert chain is DeviceCert model;
```

Comment:
SPDM 1.3 introduces a third certificate model, generic certificates. In anticipation of that this parameter should be something like `uint8_t cert_model`. @jyao1 do you agree?

Reply:
If we need consider SPDM 1.3, yes, change to uint8_t.

Also, we need to add spdm_version, right?

Reply:
do you mean as another argument to this function? if so why?

Reply:
> SPDM 1.3 introduces a third certificate model, generic certificates. In anticipation of that this parameter should be something like `uint8_t cert_model`. @jyao1 do you agree?

I feel like this should be a separate PR? There's a few other function that need to be updated, and probably the addition of new macros for each type also.

Reply:
>Also, we need to add spdm_version, right?

In this particular case no, but it might not hurt in the long-run.

>do you mean as another argument to this function?

No, it would replace `is_device_cert_model`.

>if so, why?

So we won't need `libspdm_verify_cert_chain_data_ex2` when we start to implement 1.3.

Reply:
I can add this change, but we will need to change the api for libspdm_x509_set_cert_certificate_check() and all subsequent calls to accommodate the `uint8_t cert_model`. Is that still appropriate for this PR? or is it more appropriate to just translate the u8 into a bool for now within `_data_ex()`?